### PR TITLE
GH-1898 - ParquetStore validates snapshot folder does not already exist

### PR DESCRIPTION
**Pull request by benj-fletch**
Enhancement to `ParquetStore#setLatestSnapshot()` to ensure that the snapshot folder being set to does not already exist

### CODE REVIEW

**Comment by m55624** [comment]
Hi @benj-fletch, thanks very much for the contribution.  I have just noticed the Travis-CI build has failed on your test, this will need to be fixed before it can be merged, thanks!

**Comment by benj-fletch** [comment]
thanks @m55624, I am just resolving the issue; apologies I was a little overzealous in opening the PR!

**Comment by m55624** [comment]
Hi @benj-fletch, just had a read quickly and the ticket is to check the snapshot path does exist, however your fix currently checks it does not exist.  Just on another note you need to use an if round the exists method, this will currently fail on an IOException which is not how you want to test the existence of the directory.  Let me know if you need anymore direction and I can add some change suggestions.

**Comment by benj-fletch** [comment]
Hi @m55624, I have updated the PR with a change that actually fixes the ticket. I should have read it more carefully the first time.

I've also added an extra positive test for the method

**Comment by m55624** [comment]
@benj-fletch thanks very much for that, looks great.  Just one small issue with the IOException catch then it should be good to go.  Thanks again for the contribution :)

File: store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/ParquetStore.java [hunk 1]
```diff
@@ -405,7 +405,15 @@ public long getLatestSnapshot() throws StoreException {
...
 
     public void setLatestSnapshot(final long snapshot) throws StoreException {
-        // TODO validate that a directory with this name exists
+        final Path snapshotPath = new Path(getDataDir(), getSnapshotPath(snapshot));
+        try {
+            if (!fs.exists(snapshotPath)) {
+                throw new StoreException(String.format("Failed setting currentSnapshot: '%s' does not exist", snapshotPath.toString()));
+            }
+        } catch (final IOException e) {
+            throw new StoreException(String.format("Failed setting currentSnapshot: '%s' cannot be found", getDataDir()), e);
```
[the remark this answers is not in the document]
Comment (benj-fletch) [reply store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/ParquetStore.java]: Yep, have pushed a change for it just now

**Comment by benj-fletch** [comment]
@m55624 No worries, I have made the change. Thanks for taking the time to look at it. I am hoping to contribute more in future and will have a look at other tickets once this one is resolved.

**Comment by gaffer01** [comment]
@benj-fletch Thanks for your contribution.

**Pull request merged**